recipes/sdbus-cpp/all/conanfile.py

```diff
@@ -48,74 +42,90 @@ def _minimum_compilers_version(self):
+        tc.variables["BUILD_DOC"] = False
+        tc.variables["BUILD_TESTS"] = False
+        tc.variables["BUILD_LIBSYSTEMD"] = False
+        tc.generate()
+
+        # workaround for https://gitlab.kitware.com/cmake/cmake/-/issues/18150
+        copy(self, "*.pc", self.generators_folder,
+             os.path.join(self.generators_folder, "lib", "pkgconfig"))
 
     def build(self):
-        self._patch_sources()
-        cmake = self._configure_cmake()
+        apply_conandata_patches(self)
+        cmake = CMake(self)
+        cmake.configure()
         cmake.build()
 
     def package(self):
-        cmake = self._configure_cmake()
+        cmake = CMake(self)
+        cmake.configure()
         cmake.install()
-        self.copy(pattern="COPYING", dst="licenses", src=self._source_subfolder)
-        tools.rmdir(os.path.join(self.package_folder, "lib", "pkgconfig"))
-        tools.rmdir(os.path.join(self.package_folder, "lib", "cmake"))
+        copy(self, "COPYING*", self.source_folder,
+             os.path.join(self.package_folder, "licenses"))
+        rmdir(self, os.path.join(self.package_folder, "lib", "pkgconfig"))
+        rmdir(self, os.path.join(self.package_folder, "lib", "cmake"))
 
     def package_info(self):
+        self.cpp_info.set_property("cmake_file_name", "sdbus-c++")
+        self.cpp_info.components["sdbus-c++"].libs = ["sdbus-c++"]
+        self.cpp_info.components["sdbus-c++"].requires.append(
+            "libsystemd::libsystemd")
+        self.cpp_info.components["sdbus-c++"].set_property(
+            "cmake_target_name", "SDBusCpp::sdbus-c++")
+        self.cpp_info.components["sdbus-c++"].set_property(
+            "pkg_config_name", "sdbus-c++")
+
+        if self.options.with_code_gen:
+            bin_path = os.path.join(self.package_folder, "bin")
+            self.output.info("Appending PATH env var with : {}".format(bin_path))
+            self.buildenv_info.prepend_path("PATH", bin_path)
+
+        # TODO: to remove in conan v2 once cmake_find_package_* generators removed
         self.cpp_info.names["cmake_find_package"] = "SDBusCpp"
         self.cpp_info.names["cmake_find_package_multi"] = "SDBusCpp"
         self.cpp_info.filenames["cmake_find_package"] = "sdbus-c++"
         self.cpp_info.filenames["cmake_find_package_multi"] = "sdbus-c++"
-        self.cpp_info.components["sdbus-c++"].libs = ["sdbus-c++"]
-        self.cpp_info.components["sdbus-c++"].requires.append("libsystemd::libsystemd")
         self.cpp_info.components["sdbus-c++"].names["cmake_find_package"] = "sdbus-c++"
         self.cpp_info.components["sdbus-c++"].names["cmake_find_package_multi"] = "sdbus-c++"
         self.cpp_info.components["sdbus-c++"].names["pkg_config"] = "sdbus-c++"
-        if self.options.with_code_gen:
-            bin_path = os.path.join(self.package_folder, "bin")
-            self.output.info("Appending PATH env var with : {}".format(bin_path))
-            self.env_info.PATH.append(bin_path)
```

Comment:
Keep env_info for compatibility with Conan 1.x. Otherwise, you will break users.

Reply:
Thanks. I probably missed env_info during cleaning the code up. Added it back.